analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -0,0 +1,129 @@
+        ) = GoMod(managerName, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    override fun command(workingDir: File?): String {
+        return "go"
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class ModuleDependency(
+        @JsonProperty("Path") val name: String,
+        @JsonProperty("Dir") val dir: String,
+        @JsonProperty("Version") val version: String?,
+        @JsonProperty("Main") val isMain: Boolean = false
+    )
+
+    override fun resolveDependencies(definitionFile: File): ProjectAnalyzerResult? {
+        val projectDir = definitionFile.parentFile
+
+        if (!analyzerConfig.allowDynamicVersions &&
+            !projectDir.resolve("go.sum").exists()
+        ) {
+            throw IOException("No go.sum file found in '$projectDir'.")
+        }
+
+        installDependencies(projectDir)
+
+        val modules = listModules(projectDir)
+        val (mains, dependencies) = modules.asSequence().partition { it.isMain }
+        val projectName = mains.firstOrNull()?.name ?: ""
+        val packages = dependencies.asSequence()
+            .map {
+                Package(
+                    id = Identifier(managerName, "", it.name, it.version ?: ""),
+                    declaredLicenses = sortedSetOf(), // Go mod doesn't support declared licenses
+                    description = "",
+                    homepageUrl = "",
+                    binaryArtifact = RemoteArtifact.EMPTY,
+                    sourceArtifact = RemoteArtifact.EMPTY,
+                    vcs = VcsInfo.EMPTY
+                )
+            }
+
+        val packageRefs = packages.map { it.toReference(linkage = PackageLinkage.STATIC) }
+        val scope = Scope("default", packageRefs.toSortedSet())
+
+        val projectVcs = processProjectVcs(projectDir)
+        return ProjectAnalyzerResult(
+            project = Project(
+                id = Identifier(
+                    type = managerName,
+                    namespace = "",
+                    name = projectName,
+                    version = projectVcs.revision
+                ),
+                definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                declaredLicenses = sortedSetOf(), // Go mod doesn't support declared licenses
+                vcs = projectVcs,
+                vcsProcessed = projectVcs,
+                homepageUrl = "",
+                scopes = sortedSetOf(scope)
+            ),
+            packages = packages.mapTo(sortedSetOf()) { it.toCuratedPackage() }
+        )
+    }
+
+    private fun installDependencies(workingDir: File) {
+        run("mod", "vendor", workingDir = workingDir).requireSuccess()
+        run("mod", "tidy", workingDir = workingDir).requireSuccess()
+    }
+
+    private fun listModules(projectDir: File): Iterator<ModuleDependency> {
+        val pc = run("list", "-json", "-m", "-mod=vendor", "all", workingDir = projectDir).requireSuccess()
```

Comment:
The list command produces a flat list while we're actually interrested in the dependency tree.
How about using the `graph` command instead of or in additiona to the `list` command to resemble the tree edges ?

Reply:
Yes. `graph` command should be used together with the `list`. From my point of view this is an improvement and for basic support flat structure is enough. Am I right?

Reply:
Done.